src/tt2dtm/pydantic_models/refine_template_manager.py

```diff
@@ -0,0 +1,313 @@
+"""Pydantic model for running the refine template program."""
+
+from typing import Any, ClassVar
+
+import numpy as np
+import pandas as pd
+import torch
+from pydantic import ConfigDict
+
+from tt2dtm.backend.core_refine_template import core_refine_template
+from tt2dtm.pydantic_models.computational_config import ComputationalConfig
+from tt2dtm.pydantic_models.correlation_filters import PreprocessingFilters
+from tt2dtm.pydantic_models.defocus_search import DefocusSearchConfig
+from tt2dtm.pydantic_models.formats import REFINED_DF_COLUMN_ORDER
+from tt2dtm.pydantic_models.match_template_manager import MatchTemplateManager
+from tt2dtm.pydantic_models.orientation_search import RefineOrientationConfig
+from tt2dtm.pydantic_models.particle_stack import ParticleStack
+from tt2dtm.pydantic_models.types import BaseModel2DTM, ExcludedTensor
+from tt2dtm.utils.data_io import load_mrc_volume
+
+
+class RefineTemplateManager(BaseModel2DTM):
+    """Model holding parameters necessary for running the refine template program.
+
+    Attributes
+    ----------
+    template_volume_path : str
+        Path to the template volume MRC file.
+    particle_stack : ParticleStack
+        Particle stack object containing particle data.
+    defocus_refinement_config : DefocusSearchConfig
+        Configuration for defocus refinement.
+    orientation_refinement_config : RefineOrientationConfig
+        Configuration for orientation refinement.
+    preprocessing_filters : PreprocessingFilters
+        Filters to apply to the particle images.
+    computational_config : ComputationalConfig
+        What computational resources to allocate for the program.
+    template_volume : ExcludedTensor
+        The template volume tensor (excluded from serialization).
+
+    Methods
+    -------
+    TODO serialization/import methods
+    __init__(self, skip_mrc_preloads: bool = False, **data: Any)
+        Initialize the refine template manager.
+    make_backend_core_function_kwargs(self) -> dict[str, Any]
+        Create the kwargs for the backend refine_template core function.
+    run_refine_template(self, orientation_batch_size: int = 64) -> None
+        Run the refine template program.
+    """
+
+    model_config: ClassVar = ConfigDict(arbitrary_types_allowed=True)
+
+    template_volume_path: str  # In df per-particle, but ensure only one reference
+    particle_stack: ParticleStack
+    defocus_refinement_config: DefocusSearchConfig
+    orientation_refinement_config: RefineOrientationConfig
+    preprocessing_filters: PreprocessingFilters
+    # refine_template_result: RefineTemplateResult
```

Comment:
Remove the RefineTemplateResult object. May be added back in the future for more complex workflows...